content/master/concepts/usages.md

```diff
@@ -0,0 +1,222 @@
+
+```yaml {label="order"}
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: Usage
+metadata:
+  name: release-uses-cluster
+spec:
+  of:
+    apiVersion: eks.upbound.io/v1beta1
+    kind: Cluster
+    resourceRef:
+      name: my-cluster
+  by:
+    apiVersion: helm.crossplane.io/v1beta1
+    kind: Release
+    resourceRef:
+      name: my-prometheus-chart
+```
+
+### Using selectors with usages
+
+Usages can use {{<hover label="selectors" line="9">}}selectors{{</hover>}}
+to define the resource in use or the using one.
+This enables using {{<hover label="selectors" line="12">}}labels{{</hover>}} or
+{{<hover label="selectors" line="10">}}matching controller references{{</hover>}}
+to define resource instead of providing the resource name.
+
+```yaml {label="selectors"}
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: Usage
+metadata:
+  name: release-uses-cluster
+spec:
+  of:
+    apiVersion: eks.upbound.io/v1beta1
+    kind: Cluster
+    resourceSelector:
+      matchControllerRef: false # default, and could be omitted
+      matchLabels:
+        foo: bar
+  by:
+    apiVersion: helm.crossplane.io/v1beta1
+    kind: Release
+    resourceSelector:
+       matchLabels:
+          baz: qux
+```
+
+Once the `Usage` controller resolves the selectors, it persists the resource
+name in the 
+{{<hover label="selectors-resolved" line="10">}}resourceRef.name{{</hover>}}
+field. The following example shows the `Usage` resource after the resolution of
+selectors.
+
+{{<hint "important" >}}
+<!-- vale write-good.Passive = NO -->
+The selectors are resolved only once. If there are more than one matches, a
+random resource is selected from the list of matched resources.
+<!-- vale write-good.Passive = YES -->
+{{< /hint >}}
+
+```yaml {label="selectors-resolved"}
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: Usage
+metadata:
+  name: release-uses-cluster
+spec:
+  of:
+    apiVersion: eks.upbound.io/v1beta1
+    kind: Cluster
+    resourceRef:
+       name: my-cluster
+    resourceSelector:
+      matchLabels:
+        foo: bar
+  by:
+    apiVersion: helm.crossplane.io/v1beta1
+    kind: Release
+    resourceRef:
+       name: my-cluster
+    resourceSelector:
+       matchLabels:
+          baz: qux
+```
+
+## Usage in a composition
+
+A typical use case for Usages is to define a deletion ordering between the
+resources in a Composition. The Usages support
+[matching controller reference]({{<ref "./compositions#match-a-controller-reference" >}})
+in selectors to ensures that the matching resource is in the same composite
+resource in the same way as [cross-resource referencing]({{<ref "./compositions#cross-resource-references" >}}).
+
+The following example shows a Composition that defines a deletion ordering
+between a `Cluster` and a `Release` resource. The `Usage` blocks deletion of
+the `Cluster` resource until the `Release` resource is successfully deleted.
+
```

Comment:
TODO: Add the notice here: https://github.com/crossplane/crossplane/pull/4444#issuecomment-1700988736